synapse/http/matrixfederationclient.py

```diff
@@ -715,6 +715,9 @@ def build_auth_headers(
         Returns:
             A list of headers to be added as "Authorization:" headers
         """
+        if destination is None and destination_is is None:
+            raise ValueError("destination and destination_is cannot both be None!")
```

Comment:
Did this check make sense before this change too?

Reply:
Yep

Reply:
I've added that because https://github.com/matrix-org/synapse/blob/5f0c507b8922463d1ae98e9f92a56c2c01c32c4e/synapse/http/matrixfederationclient.py#L748 would result in a key error otherwise, which is a relatively weird error message.